assets/src/dashboard/app/api/apiProvider.js

```diff
@@ -112,6 +108,25 @@ export default function ApiProvider({ children }) {
     [api.stories, editStoryURL]
   );
 
+  const fetchTemplates = useCallback(() => {
+    const reshapedTemplates = getAllTemplates({ pluginDir }).map(
+      reshapeTemplateObject
+    );
+    setTemplates(reshapedTemplates);
+
+    return Promise.resolve(reshapedTemplates);
+  }, [pluginDir]);
+
+  const fetchTemplate = useCallback(
+    async (id) => {
+      const fetchedTemplates = await fetchTemplates();
+      return Promise.resolve(
+        fetchedTemplates.find((template) => template.id === id)
+      );
+    },
+    [fetchTemplates]
+  );
```

Comment:
Even though we're pulling the templates from json files, I decided to structure the apiProvider as if they were coming from api requests.  This way when we get the real api endpoints, the underlying code that uses the apiProvider won't have to change.  

(Note:  A bigger refactor of the apiProvider to stub out all the needed template api endpoints is coming in another PR.)